debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/stackwalker/StackWalker.java

```diff
@@ -139,116 +148,157 @@ private static IStackWalker getImpl()
+				if (walkState.arg0EA == null) {
+					walkState.arg0EA = walkThread.arg0EA();
+				}
+				if (walkState.pc == null) {
+					walkState.pc = walkThread.pc();
+				}
+				walkState.pcAddress = walkThread.pcEA();
+				if (walkState.walkSP == null) {
+					walkState.walkSP = walkThread.sp();
+				}
+				if (walkState.literals == null) {
+					walkState.literals = walkThread.literals();
+				}
+				if (walkState.walkedEntryLocalStorage == null) {
+					walkState.walkedEntryLocalStorage = walkThread.entryLocalStorage();
+				}
+				walkState.j2iFrame = walkThread.j2iFrame();
+				walkState.decompilationStack = walkThread.decompilationStack();
+			} catch (CorruptDataException ex) {
+				raiseCorruptDataEvent("CDE thrown extracting initial stack walk state. walkThread = " + walkThread.getHexAddress(), ex, true);
+				return StackWalkResult.STACK_CORRUPT;
+			}
+			return walkStackFrames(walkState);
+		}
+
+		private Object getField(Lookup lookup, Object obj, Class<?> refc, String name, Class<?> rtype) throws Throwable {
+			MethodHandle mh = lookup.findVirtual(refc, name, MethodType.methodType(rtype));
+			return mh.invoke(obj);
+		}
+
+		@Override
+		public StackWalkResult walkStackFrames(WalkState walkState, long continuationAddress) {
 			try {
-				return walkStackFrames(walkState, 
-						walkState.walkThread.sp(), 
-						walkState.walkThread.arg0EA(), 
-						walkState.walkThread.pc(), 
-						walkState.walkThread.literals(),
-						walkState.walkThread.entryLocalStorage());
+				Class<?> contClass = Class.forName("com.ibm.j9ddr.vm29.pointer.generated.J9VMContinuationPointer");
+				Lookup lookup = MethodHandles.lookup();
+				MethodHandle mhCast = lookup.findStatic(contClass, "cast", MethodType.methodType(contClass, long.class));
+				Object cont = mhCast.invoke(continuationAddress);
+
+				walkState.threadAddress = (long)getField(lookup, cont, contClass, "getAddress", long.class);
+				walkState.arg0EA = (UDATAPointer)getField(lookup, cont, contClass, "arg0EA", UDATAPointer.class);
+				walkState.pc = (U8Pointer)getField(lookup, cont, contClass, "pc", U8Pointer.class);
+				walkState.pcAddress = (PointerPointer)getField(lookup, cont, contClass, "pcEA", PointerPointer.class);
+				walkState.walkSP = (UDATAPointer)getField(lookup, cont, contClass, "sp", UDATAPointer.class);
+				walkState.literals  = (J9MethodPointer)getField(lookup, cont, contClass, "literals", J9MethodPointer.class);
+				walkState.privateFlags  = new UDATA(0);
+				walkState.j2iFrame = (UDATAPointer)getField(lookup, cont, contClass, "j2iFrame", UDATAPointer.class);
+				walkState.decompilationStack = (J9JITDecompilationInfoPointer)getField(lookup, cont, contClass, "decompilationStack", J9JITDecompilationInfoPointer.class);
+
+				walkState.javaVM = J9RASHelper.getVM(DataType.getJ9RASPointer());
+				J9VMEntryLocalStoragePointerWrapper wrapperELS = new J9VMEntryLocalStoragePointerWrapper(cont);
+				walkState.walkedEntryLocalStorage = wrapperELS;
```

Comment:
> This should use `J9VMContinuationPointer` directly by adding in `AuxFieldInfo29.dat` declarations of the fields 

The VM structure `struct J9VMContinuation` is only defined for Java 19+, including it directly will cause compile issues for Java 8/11/17
> can we replace the field in `WalkState` with the necessary components instead?

The `J9VMEntryLocalStoragePointer` is using a linked-list structure, we would not be able to directly place the fields into `WalkState` without heavily modifying the existing code to account for how to iterate the list

Reply:
The additions I listed for `AuxFieldInfo29.dat` will make that type available to DDR_VM code.

The approach using `J9VMEntryLocalStoragePointerWrapper` just seems wrong to me. Perhaps `Supplier<J9VMEntryLocalStoragePointer>` is the interface we should use. The variant for a thread reads the field from the thread, and the variant for a continuation gets it via a `J9VMContinuationPointer`.

Reply:
> The additions I listed for `AuxFieldInfo29.dat` will make that type available to DDR_VM code.

I wasn't sure if adding type after field declarations would make them available to DDR, if that is the case, we can switch to direct access under a try/catch block when reading the fields.
> The approach using `J9VMEntryLocalStoragePointerWrapper` just seems wrong to me. Perhaps `Supplier<J9VMEntryLocalStoragePointer>` is the interface we should use. The variant for a thread reads the field from the thread, and the variant for a continuation gets it via a `J9VMContinuationPointer`.

The reasoning behind using `J9VMEntryLocalStoragePointerWrapper` is to minimize the code changes on existing stack walker code and to keep it as similar to the VM's walkStackFrames() API as possible to make future maintenance easier. Changing to use `Supplier<T>` interface would require refactor on all code access to ELS right?

Reply:
I don't see that `J9VMEntryLocalStoragePointerWrapper` is needed at all (nor is using `Supplier<?>`).

`StackWalker` can simply use this for a continuation:
```
    walkState.walkedEntryLocalStorage = continuation.oldEntryLocalStorage();
````

Reply:
> I don't see that `J9VMEntryLocalStoragePointerWrapper` is needed at all (nor is using `Supplier<?>`).
> 
> `StackWalker` can simply use this for a continuation:
> 
> ```
>     walkState.walkedEntryLocalStorage = continuation.oldEntryLocalStorage();
> ```

The oldEntryLocalStorage is the list that is previous the current entry, the J9VMContinuation stores the necessary fields from the current ELS to avoid creating ELS outside of callin code as the location of each ELS struct is relative to the cInterpreter frame. Hence when DDR is trying to read the ELS data from a J9VMContinuation, the current entry must be recreated as there isn't an actual pointer in the core for that entry.

Reply:
Then it seems you want a structure that provides `J9VMEntryLocalStorage* and those other "necessary fields".

I don't think it's appropriate to have something that extends `J9VMEntryLocalStoragePointe` (i.e. `J9VMEntryLocalStoragePointerWrapper`) as that's not an accurate representation of the world.

Reply:
> Then it seems you want a structure that provides `J9VMEntryLocalStorage* and those other "necessary fields".

I want a structure that provides the "necessary fields" from a `J9VMEntryLocalStorage*` without a actual `J9VMEntryLocalStorage` structure existing in the corefile. (these fields are stored in `J9VMContinuation`)

In corefile:
```
J9VMEntryLocalStorage*                         J9VMContinuation*
    -> fieldA (necessary)                             -> fieldA
    -> fieldB (necessary)                             -> fieldB
    -> fieldC                                         -> oldELS (J9VMEntryLocalStorage*) 
    -> oldELS (J9VMEntryLocalStorage*)
```
in this case, the structure should be able to provide `fieldA / fieldB` as if they are from a `J9VMEntryLocalStorage`
Using the `J9VMEntryLocalStoragePointerWrapper` class would avoid having to rewrite all locations where `J9VMEntryLocalStorage` is accessed in the StackWalker code

Reply:
Perhaps the right container is `WalkState`: we either initialize that state from a thread or a continuation. Advancing from one stack frame to the next is something that should be associated with that state.

Reply:
Yes, I suggested an interface that would allow access to ELS and whatever other information needed to walk stacks. That interface is not ELS, but it seems reasonable to me to adjust WalkState to capture what is needed so navigation from one stack frame to the next can occur. Rather than require two pieces of information (WalkState and something else), the stack walker would use one (WalkState). I think part of the difficulty is that the fields of WalkState are public and modified all over the place - a mistake in my opinion. Rather than perpetuate and expand on that mistake, I'm suggesting we take the time to improve the situation (perhaps even the native implementation as well).

Reply:
I think we should try to have this version merged as a baseline on meeting functional expectation.

Refactoring WalkState to an improved design of how stack walking is done should be pursued as a new issue. I don't want to delay the loom features due to redesign / test on a rework of DDR stack walker (potentially even the native impl in swalk.c).

Reply:
I disagree. From my perspective, this is not ready, and I don't think this DDR portion interferes with meeting functional expectations. It can wait until we have an appropriate implementation.